tools/backups2datalad/adandi.py

```diff
@@ -169,6 +169,7 @@ async def aget_assets(self) -> AsyncGenerator[RemoteAsset, None]:
             self.aclient.paginate(
                 f"{self.version_api_path}assets/",
                 params={"order": "created", "metadata": "1", "page_size": "1000"},
+                timeout=60,
```

Comment:
I don't think it worked since
```
2022-10-03T17:02:20-0400 [WARNING ] dandi: A newer version (0.46.3) of dandi/dandi-cli is available. You are using 0.46.2
2022-10-03T17:04:22-0400 [WARNING ] backups2datalad: Retrying GET request to https://api.dandiarchive.org/api/dandisets/000108/versions/draft/assets/?metadata=1&order=created&page=6&page_size=1000 in 0.974346 seconds as it raised ConnectTimeout:
2022-10-03T17:04:56-0400 [WARNING ] backups2datalad: Retrying GET request to https://api.dandiarchive.org/api/dandisets/000108/versions/draft/assets/?metadata=1&order=created&page=6&page_size=1000 in 2.025987 seconds as it raised ConnectTimeout:
2022-10-03T17:05:30-0400 [WARNING ] backups2datalad: Retrying GET request to https://api.dandiarchive.org/api/dandisets/000108/versions/draft/assets/?metadata=1&order=created&page=6&page_size=1000 in 3.988392 seconds as it raised ConnectTimeout:
2022-10-03T17:06:00-0400 [WARNING ] backups2datalad: Retrying GET request to https://api.dandiarchive.org/api/dandisets/000108/versions/draft/assets/?metadata=1&order=created&page=6&page_size=1000 in 8.029622 seconds as it raised ConnectTimeout:
2022-10-03T17:06:27-0400 [WARNING ] backups2datalad: Retrying GET request to https://api.dandiarchive.org/api/dandisets/000108/versions/draft/assets/?metadata=1&order=created&page=6&page_size=1000 in 15.438817 seconds as it raised ConnectTimeout:
```
so we get requests repeated faster than 6 - so timeout must have happened before 60 seconds :-/

Reply:
well, there is also `aget_asset_by_path` function which goes to the same end point but since it is for a single `path` I do not think it is the one used..

Reply:
oh -- we need a following patch to pass timeout into the next page `get`:
```
diff --git a/tools/backups2datalad/adandi.py b/tools/backups2datalad/adandi.py
index c159359..00875a7 100644
--- a/tools/backups2datalad/adandi.py
+++ b/tools/backups2datalad/adandi.py
@@ -76,7 +76,7 @@ class AsyncDandiClient(AsyncResource):
             for item in r["results"]:
                 yield item
             if r.get("next"):
-                r = await self.get(r["next"])
+                r = await self.get(r["next"], **kwargs)
             else:
                 break
 
```
I will apply it directly and see